stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Any, Final, Literal, TypeVar, final
+from typing_extensions import Buffer, Self, TypeAlias
+
+_T = TypeVar("_T")
+_SendType: TypeAlias = Literal["send", "recv", "both"]
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+@final
```

Comment:
Hmm, it seems like it's subclassable?

```pycon
>>> class Foo(ChannelID): pass
...
>>>
```

But on the other hand, given that the constructor raises `TypeError`, I guess there's no good reason to subclass it. Maybe add a comment saying that's why we mark it as `@final`, if that's indeed why you have marked it as `@final`? :-)

Reply:
Yeah, basically that, I'll leave a comment clarifying.

Reply:
https://github.com/python/typeshed/pull/12199/commits/cf8580e6c61a1f5877605cb6bf1a685ef72bc9e2

Reply:
thanks!